storey/sources.py

```diff
@@ -666,7 +666,11 @@ async def _run_loop(self):
                     else:
                         line_id = self._get_uuid()
                     element = self._get_element(body, columns)
-                    keys = keys[0] if len(keys) == 1 else (None if not keys else keys)
+                    keys = (
+                        keys[0]
+                        if len(keys) == 1 and not isinstance(self._key_field, list)
+                        else (None if not keys else keys)
+                    )
```

Comment:
Can we simplify and clarify as follows?
```suggestion
                    if len(keys) == 0:
                        keys = None
                    elif not isinstance(self._key_field, list):
                        keys = keys[0]
```